snuba/datasets/__init__.py

```diff
@@ -33,12 +39,55 @@ def get_writer(self, options=None, table_name=None):
         from snuba import settings
         from snuba.clickhouse.http import HTTPBatchWriter
 
+        def default(value):
+            if isinstance(value, datetime):
+                return value.strftime(DATETIME_FORMAT)
+            else:
+                raise TypeError
+
+        def encode(rows: Iterable[WriterTableRow]) -> Iterable[bytes]:
+            return map(
+                lambda row: json.dumps(row, default=default).encode("utf-8"),
```

Comment:
Is there any reason that you know of to _not_ make the switch here right now, other than just decreasing surface area for this specific change? (If you don't want to do it right now that's fine, but I'd like both write paths to share the same encoding as soon as we can.)

Reply:
which switch are you talking about ?

Reply:
The switch to `rapidjson` instead of standard library `json`.

Reply:
It is to reduce the impact of this change.
I think rapidjson should be eventually there everywhere, but we should probably add it use case per use case. 
As an example, using it on groupedmessage is trivial since there are 6 fields and 3 are dates we treat as strings, we do not even make it serialize datetime objects. 
Introducing that on events would be a different matter we should probably test it out on a traffic that is not production first.

Reply:
nevermind. I removed the distinction